pkg/appmanager/appManager.go

```diff
@@ -1110,20 +1110,28 @@ func (appMgr *Manager) syncRoutes(
 		if route.ObjectMeta.Namespace != sKey.Namespace {
 			continue
 		}
+
+		//FIXME(kenr): why do we process services that aren't associated
+		//             with a route?
+		svcName := getRouteCanonicalService(route)
+		if existsRouteServiceName(route, sKey.ServiceName) {
+			svcName = sKey.ServiceName
+		}
+
 		if nil != route.Spec.TLS {
 			switch route.Spec.TLS.Termination {
 			case routeapi.TLSTerminationPassthrough:
-				updateDataGroupForPassthroughRoute(route, DEFAULT_PARTITION,
+				updateDataGroupForPassthroughRoute(route, svcName, DEFAULT_PARTITION,
 					sKey.Namespace, dgMap)
 			case routeapi.TLSTerminationReencrypt:
-				updateDataGroupForReencryptRoute(route, DEFAULT_PARTITION,
+				updateDataGroupForReencryptRoute(route, svcName, DEFAULT_PARTITION,
```

Comment:
Same comment for passthrough routes applies here as well for re-encrypt routes.

Reply:
Same as above.